### ISSUE TITLE
github script variable conflicting with user script

### ISSUE THREAD

**Issue by y-nk** [actions/github-script#343]
**Describe the bug**
it seems there's a crash between the script we pass and the code defined in here

**To Reproduce**
1. Create a script with the value: `script: const exec = 'foo'`
2. See error "exec already defined"

**Expected behavior**
there should not be javascript error

**Comment by maxkostow** [comment]
My failed action pulled `SHA:98814c53be79b1d30f795b907e553d8679345975` while my previous successful action pulled `SHA:d556feaca394842dc55e4734bf3bb9f685482fa0`

**Comment by joshmgross** [comment]
I don't think there's an easy way to solve this, I'd recommend avoiding conflicts with the built-in arguments https://github.com/actions/github-script#actionsgithub-script

For new arguments we add in the future (like `fetch`), I'll make sure we do a major version update.

**Comment by y-nk** [comment]
@joshmgross i dont think passing @actions/exec (or fetch) is worth it.

https://github.com/actions/github-script/blob/98814c53be79b1d30f795b907e553d8679345975/src/main.ts#L51-L64

a consumer can always do `npm install @actions/exec` in a step before and use require if they want to (which is the documented way in the README).

doing so is an assumption that it's useful, but it's rather adding bundle size on this action when people can load this by themselves. going this way, soon enough loading the entire lodash could also be seen as useful and bloat the action in the long run.

TLDR: as long as it's not useful for this particular action, it'd be nicer not to load anything else than `context` and `github` to the async function 🙏

**Comment by joshmgross** [comment]
Thanks for the feedback @y-nk.

This action is meant as a general purpose solution to quickly write JS scripts in an Actions workflow. Everyone is going to have different use-cases, so it's difficult to define what's "useful for this particular action". `fetch` was mostly added because it's generally useful and we're stuck on Node 16 for now, it won't be an extra dependency on newer Node versions.

If you have a very specific set of dependencies and want a small bundle size, I'd recommend writing a regular [Javascript action](https://docs.github.com/en/actions/creating-actions/creating-a-javascript-action) so that you'll have full control over that.

**Comment by y-nk** [comment]
@joshmgross thanks for the explanation and the suggestion :)
